File: Easy/2026-02-25-1356-Sort-Integers-by-The-Number-of-1-Bits/solution.js
var sortByBits = function(arr) {
    return arr.sort((a, b) => {
        const countA = a.toString(2).split('0').join('').length;
        const countB = b.toString(2).split('0').join('').length;
        return countA === countB ? a - b : countA - countB;
    });
};

File: Easy/2026-02-25-1356-Sort-Integers-by-The-Number-of-1-Bits/solution.java
import java.util.Arrays;

class Solution {
    public int[] sortByBits(int[] arr) {
        Arrays.sort(arr, (a, b) -> {
            int countA = Integer.bitCount(a);
            int countB = Integer.bitCount(b);
            if (countA == countB) {
                return Integer.compare(a, b);
            }
            return Integer.compare(countA, countB);
        });
        return arr;
    }
}